homeassistant/components/vera/lock.py

```diff
@@ -59,6 +60,22 @@ def unlock(self, **kwargs: Any) -> None:
         self.vera_device.unlock()
         self._state = STATE_UNLOCKED
 
+    def setpin(self, **kwargs: Any) -> None:
+        """Set Pin on the device."""
+        result = self.vera_device.set_new_pin(name=kwargs.name, pin=kwargs.pin)
+        if result.status_code==200:
+            self._state = STATE_OK
+        else:
+            self._state = STATE_ERROR
```

Comment:
These are not valid states for the lock.

Also, if we succeed or fail to update the pin on the lock, the lock stays locked or unlocked, it doesn't enter an OK or error state?

Reply:
That is true, the lock stays locked or unlocked irrespective of the set/clear pin operation.

What would be the proposed way to intimate the user about the pin operation's error/success if we don't want to use the state? Add another attribute?